src/rust/src/x509/common.rs

```diff
@@ -520,34 +521,63 @@ pub(crate) fn datetime_to_py_utc<'p>(
         dt.hour(),
         dt.minute(),
         dt.second(),
-        0,
+        microseconds,
         Some(&pyo3::types::PyTzInfo::utc(py)?.to_owned()),
     )?;
     Ok(py_datetime.into_any())
 }
 
-pub(crate) fn py_to_datetime(
+pub(crate) fn datetime_to_py_utc<'p>(
+    py: pyo3::Python<'p>,
+    dt: &asn1::DateTime,
+) -> pyo3::PyResult<pyo3::Bound<'p, pyo3::PyAny>> {
+    datetime_to_py_utc_with_microseconds(py, dt, 0)
+}
+
+// Convert Python's datetime objects to a tuple of `asn1::DateTime` and
+// microseconds.
+pub(crate) fn py_to_datetime_with_microseconds(
     py: pyo3::Python<'_>,
-    val: pyo3::Bound<'_, pyo3::types::PyDateTime>,
-) -> pyo3::PyResult<asn1::DateTime> {
+    val: &pyo3::Bound<'_, pyo3::types::PyDateTime>,
+) -> pyo3::PyResult<(asn1::DateTime, Option<u32>)> {
     // We treat naive datetimes as UTC times, while aware datetimes get
     // normalized to UTC before conversion.
+    let normalized: pyo3::Bound<'_, pyo3::types::PyAny>;
     let val_utc = if val.get_tzinfo().is_none() {
-        val.into_any()
+        val.as_any()
     } else {
         let utc = pyo3::types::PyTzInfo::utc(py)?;
-        val.call_method1(pyo3::intern!(py, "astimezone"), (utc,))?
+        normalized = val.call_method1(pyo3::intern!(py, "astimezone"), (utc,))?;
+        &normalized
     };
```

Comment:
I'm not sure I understand the intent here.

Reply:
yeah this was me trying to workaround the fact that I changed the parameter type from a `Bound` to a `&Bound`, but ended up making things more complicated than needed.

I made it a `Bound` and now the logic is the same as before.